src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,748 @@
         {
-            resultPtr->_LoadFragmentExtensions();
+            if (profile->Name() == preferredPowershellProfile)
+            {
+                guid = profile->Guid();
+                break;
+            }
         }
-        CATCH_LOG();
 
-        if (!fileHasData)
+        userSettings.globals->DefaultProfile(guid);
+    }
+
+    // 2.
+    {
+        for (const auto& profile : userSettings.profiles)
         {
-            // We didn't find the user settings. We'll need to create a file
-            // to use as the user defaults.
-            // For now, just parse our user settings template as their user settings.
-            auto userSettings{ resultPtr->_ApplyFirstRunChangesToSettingsTemplate(UserSettingsJson) };
-            resultPtr->_ParseJsonString(userSettings, false);
-            needToWriteFile = true;
+            if (profile->Guid() == DEFAULT_COMMAND_PROMPT_GUID)
+            {
+                profile->Name(RS_(L"CommandPromptDisplayName"));
+                break;
+            }
         }
+    }
+}
 
-        try
+// Adds profiles from .inboxSettings as parents of matching profiles in .userSettings.
+// That way the user profiles will get appropriate defaults from the generators (like icons and such).
+// If a matching profile doesn't exist yet in .userSettings, one will be created.
+void SettingsLoader::MergeInboxIntoUserSettings()
+{
+    for (const auto& generatedProfile : inboxSettings.profiles)
+    {
+        if (const auto [it, inserted] = userSettings.profilesByGuid.emplace(generatedProfile->Guid(), generatedProfile); !inserted)
         {
-            // See microsoft/terminal#2325: find the defaultSettings from the user's
-            // settings. Layer those settings upon all the existing profiles we have
-            // (defaults and dynamic profiles). We'll also set
-            // _userDefaultProfileSettings here. When we LayerJson below to apply the
-            // user settings, we'll make sure to use these defaultSettings _before_ any
-            // profiles the user might have.
-            resultPtr->_ApplyDefaultsFromUserSettings();
-
-            // Apply the user's settings
-            resultPtr->LayerJson(resultPtr->_userSettings);
+            // If inserted is false, we got a matching user profile with identical GUID.
+            // --> The generated profile is a parent of the existing user profile.
+            it->second->InsertParent(generatedProfile);
         }
-        catch (...)
+        else
         {
-            _CatchRethrowSerializationExceptionWithLocationInfo(resultPtr->_userSettingsString);
+            // If inserted is true, then this is a generated profile that doesn't exist in the user's settings.
+            // While emplace() has already created an appropriate entry in .profilesByGuid, we still need to
+            // add it to .profiles (which is basically a sorted list of .profilesByGuid's values).

```

Comment:
AH okay we CreateChild here because _there wasn't an entry in the user's settings for this default profile_. The child now acts like it belongs in the user settings, so when we write the file back out, we'll actually serialize that child to `settings.json`. Okay, that now makes some sense. Had to read this and the fragments version a few times to figure out why we were making a child here